sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/DirectStreamObserver.java

```diff
@@ -59,9 +60,8 @@ public DirectStreamObserver(Phaser phaser, CallStreamObserver<T> outboundObserve
 
   @Override
   public void onNext(T value) {
-    numberOfMessagesBeforeReadyCheck += 1;
-    if (numberOfMessagesBeforeReadyCheck >= maxMessagesBeforeCheck) {
-      numberOfMessagesBeforeReadyCheck = 0;
+    if (maxMessagesBeforeCheck <= 1 ||
```

Comment:
Does `maxMessagesBeforeCheck` need to be an atomic integer?

Reply:
No since it is final